server/cluster/cluster.go

```diff
@@ -464,16 +461,12 @@ func (c *RaftCluster) Stop() {
 
 // IsRunning return if the cluster is running.
 func (c *RaftCluster) IsRunning() bool {
-	c.RLock()
-	defer c.RUnlock()
-	return c.running
+	return atomic.LoadInt64(&c.running) == 1
 }
 
 // Context returns the context of RaftCluster.
 func (c *RaftCluster) Context() context.Context {
-	c.RLock()
-	defer c.RUnlock()
-	if c.running {
+	if c.IsRunning() {
```

Comment:
I'm worried that this may cause some unexpected panics. Since the PR makes the `c.running` become zero earlier than before. Would you like to take a deep dive into this to see if this will happen?

Reply:
I think it's ok. coz the context of the cluster is used for the `syncHistoryRegion` function, and if the `c.running` has been set to zero before canceling the context, it will not return the context. And if the context has been cancelled, the `RegionSyncer` will stop syncing historic regions.